mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,456 @@
+        (intervals['strain'] == strain) & (intervals['mouse'] == mouse)]
+    days = sorted(np.unique(mouse_data['day']))
+
+    # build data frame
+    bin_count = int(24 * 60 / bin_width)
+    time_movements = np.repeat(0.0, bin_count * len(days))
+    bin_length = bin_width * 60
+    for j in days:
+        M = data.load_movement(strain, mouse, day=int(j))
+        distance_df = pd.DataFrame({"start": M["t"].values[0:-1],
+                                    "end": M["t"].values[1:],
+                                    "distance":
+                                    np.linalg.norm(M[["x", "y"]].values[1:] -
+                                                   M[["x", "y"]].values[0:-1],
+                                                   axis=1)})
+        start_end = data.load_start_time_end_time(strain, mouse, j)
+        start = np.asarray(distance_df['start']) - start_end[0]
+        end = np.asarray(distance_df['end']) - start_end[0]
+        dist = distance_df['distance']
+        for i in range(len(start)):
+            start_time = start[i]
+            end_time = end[i]
+            start_index = int(start_time / (bin_width * 60))
+            end_index = int(end_time / (bin_width * 60))
+            if start_index == end_index:
+                time_movements[start_index + j *
+                               bin_count] += dist[i]
+            else:
+                time_movements[
+                    end_index + j * bin_count] += end_time % \
+                    bin_length / (end_time - start_time) * dist[i]
+                time_movements[
+                    start_index + j * bin_count] += dist[i] - \
+                    end_time % bin_length / (end_time - start_time) * dist[i]
+    ts = pd.Series(time_movements, index=pd.date_range(
+        '01/01/2014', periods=len(time_movements),
+        freq=str(bin_width) + 'min'))
+    return(ts)
+
+
+def aggregate_data(feature, bin_width):
+    r"""
+    Aggregate all the strains and mouses with any feature together
+    in one dataframe. It combines the results you got from
+    aggregate_movements and aggregate_interval. It will return
+    a dataframe with three variables: mouse, strain, feature and hour.
+
+    Parameters
+    ----------
+    feature :
+        {"AS", "F", "IS", "M_AS", "M_IS", "W"}
+    bin_width : int
+        Number of minutes, the time interval for data aggregation.
+
+    Returns
+    -------
+    pandas.dataframe
+    describe :
+        Column 0: the mouse number (number depends on strain)(0-3)
+        Column 1: the strain of the mouse (0-2)
+        Column 2: hour(numeric values below 24 accourding to bin_width)
+        Column 3: feature values
+
+    Examples
+    --------
+    >>> test = aggregate_data("Distance",20)
+    >>> print(np.mean(test["Distance"]))
+    531.4500177747973
+    """
+
+    if feature not in INTERVAL_FEATURES:
+        raise ValueError(
+            'Input value must in {"AS", "F", "M_AS", "M_IS", "W"}')
```

Comment:
@richiefeng @HaoLyuKrystal  Here in the aggregate_data function, I think Hao can deal with the feature = 'Distance'. Thus the raise ValueError should include 'Distance' in it, not just " Input value must in {"AS", "F", "M_AS", "M_IS", "W"}') ".   Now it will pump out an error notice.